ckan/logic/action/get.py

```diff
@@ -1903,21 +1907,25 @@ def package_search(context, data_dict):
         # Add them back so extensions can use them on after_search
         data_dict['extras'] = extras
 
-        for package in query.results:
-            # get the package object
-            package_dict = package.get(data_source)
-            ## use data in search index if there
-            if package_dict:
-                # the package_dict still needs translating when being viewed
-                package_dict = json.loads(package_dict)
-                if context.get('for_view'):
-                    for item in plugins.PluginImplementations(
-                            plugins.IPackageController):
-                        package_dict = item.before_view(package_dict)
-                results.append(package_dict)
-            else:
-                log.error('No package_dict is coming from solr for package '
-                          'id %s', package['id'])
+        if data_dict.get('fl') in ['id', 'name']:
+            for package in query.results:
+                results.append( {data_dict.get('fl'):package} )
```

Comment:
I noticed in ckan/ckan/lib/search/query.py class PackageSearchQuery:run() :   
```
            # if just fetching the id or name, return a list instead of a dict   
            if query.get('fl') in ['id', 'name']:   
                self.results = [r.get(query.get('fl')) for r in self.results]  
```
so we can be consistent with this?